repo/badgerds.go

```diff
@@ -0,0 +1,63 @@
+package repo
+
```

Comment:
This is IMO a good refactor, but should be done separately from the badger2 update to make reviewing easier and allowing CI to run on changes more independently.